federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -728,7 +728,9 @@ func (s *ServiceController) lockedUpdateDNSRecords(service *cachedService, clust
 	for key := range s.clusterCache.clientMap {
 		for _, clusterName := range clusterNames {
 			if key == clusterName {
-				s.ensureDnsRecords(clusterName, service)
+				if err := s.ensureDnsRecords(clusterName, service); err != nil {
+					return fmt.Errorf("Failed to update DNS records for service %v due to %v", service, err)
```

Comment:
How about logging the error, but still try to ensureDNS records for other clusters?
That way we can return errors for all clusters at once